lib/elixir/pages/Naming Conventions.md

```diff
@@ -0,0 +1,90 @@
+# Naming Conventions
+
+## Naming modules
+
+Elixir module names are written in "upper camel case", like `OptionParser`.
+
+Capital letters are kept in abbreviations, like `ExUnit.CaptureIO` or `Mix.SCM`.
```

Comment:
When you say, I want a module named `Mix.Tasks.MyApp.Cli`, Erlang will read the file `Elixir.Mix.TAsks.MyApp.Cli.beam` from the filesystem. Because filesystem may be case insensitive, it would read both `Cli.beam` and `CLI.beam`. Once the beam is read, it will check if the name you gave `Mix.Tasks.MyApp.Cli`, matches with the one in the file, which could be `Mix.Tasks.MyApp.CLI`. In case it does not match, the file won't be loaded, it will raise and an error will be logged.

Even if that did not happen I wouldn't like to add such mechanism because it is a slippery slope as we, even though, wouldn't be able to distinguish if cli means CLI, CLi or Cli, and trying all possible combinations whenever a task does not exist is too much.